post.md

```diff
@@ -0,0 +1,368 @@
+index 27aba9f59cc9..262ba2b345af 100644
+--- a/CorebootPayloadPkg/CorebootPayloadPkgIa32X64.dsc
++++ b/CorebootPayloadPkg/CorebootPayloadPkgIa32X64.dsc
+@@ -565,7 +565,6 @@ [Components.X64]
+     #------------------------------
+ 
+     <LibraryClasses>
+-      DebugLib|MdePkg/Library/UefiDebugLibConOut/UefiDebugLibConOut.inf
+       DevicePathLib|MdePkg/Library/UefiDevicePathLib/UefiDevicePathLib.inf
+       FileHandleLib|MdePkg/Library/UefiFileHandleLib/UefiFileHandleLib.inf
+       HandleParsingLib|ShellPkg/Library/UefiHandleParsingLib/UefiHandleParsingLib.inf
+```
+
+However this showed me hang in `DoShellPrompt` on function code:
+
+```
+ShellInfoObject.NewEfiShellProtocol->ReadFile(...)
+```
+
+I could not see the prompt and type any input commands.
+
+### Explaining ConIn, ConOut and ErrOut
+
+Big kudos to Laszlo Ersek who is well known from creating and maintaining OVMF.
+He pointed me to code in `ArmVirtPkg` where workaround for my problem was
+implemented.
+
+I read through code from `ArmVirtPkg/Library/PlatformBootManagerLib/PlatformBm.c` 
+and meant that I have to modify `ConIn`, `ConOut` and `ErrOut` variables. It
+was because those variables miss device path to UART device.
+
+`ConIn`, `ConOut` and `ErrOut` are global variables defined in UEFI spec. Those
+variables are available in boot time, runtime and are non volatile. This means
+that those variables are available during boot phase before firmware calls
+ExitBootServices and after that during system runtime. Those variables can be
+changed, but change takes effect after boot. So in short those variables define
+where we can find input, output and std error device.
+
+As described in [mailing thread](https://lists.01.org/pipermail/edk2-devel/2017-July/012352.html) serial
+port can be reached through `SerialPortLib` API and it worked for me during
+boot phase. Precisely what worked for me was `BaseSerialPortLib16550`. I assume
+methods in this library are not available in runtime and that's why switching
+to Shell caused no output. Second method is through `EfiSimpleTextOutProtocol`.
+Full explanation can be found in mentioned thread, but in short it is required
+to add device path of UART to mentioned global variables so those can be used.
+
+My understanding of stack is:
+
+```
+|- ShellPkg/Application/Shell/Shell.inf
+|- MdeModulePkg/Universal/Console/TerminalDxe/TerminalDxe.inf
+|- MdeModulePkg/Universal/SerialDxe/SerialDxe.inf
+|->SerialPortLib|CorebootModulePkg/Library/BaseSerialPortLib16550/BaseSerialPortLib16550.inf
+```
+
+`BaseSerialPortLib16550` works on I/O and MMIO level to initialize and provide
+read/write capability for  16550 compatible UART device. This lib is utilized
+by `SerialDxe` DXE driver. `SerialDxe` produce `gEfiSerialIoProtocolGuid` and
+`gEfiDevicePathProtocolGuid`. First abstracts any type of I/O device and
+provide communication capability for it. Second gives ability of provides
```

Comment:
`gives ability of provides` ? @pietrushnic I don't quite understand what do You mean. Maybe `Second one provides information...`?